server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -627,7 +705,7 @@ void toXContent(XContentBuilder builder, Params params, ToXContent custom) throw
     protected void serializeMappers(XContentBuilder builder, Params params) throws IOException {
         // sort the mappers so we get consistent serialization format
         Mapper[] sortedMappers = mappers.values().toArray(Mapper[]::new);
-        Arrays.sort(sortedMappers, Comparator.comparing(Mapper::name));
+        Arrays.sort(sortedMappers, Comparator.comparing(Mapper::simpleName));
```

Comment:
I have a deja-vu seeing this change. I believe I've been here before and badly broken things this way. We no longer include the full path when sorting which makes use sort inconsistently, for instance  for foo.bar and bar and baz.bar it is not clear which one comes first?

Reply:
I remember this has caused cryptic `MapperService#assertRefreshIsNotNeeded` failures when I made that change.

Reply:
Did you break it by setting it to sort by `simpleName` or  by setting it to `name`? I don't fully remember why I made this change but something was failing for me.

Reply:
I just did a quick couple of tests and nothing seemed to break when reverting this. I'll push a revert and then see what CI thinks.

Reply:
I definitely broke it with the change you had made, from name to simpleName, because name is the full path, while simple name only the leaf name. I need to rename these, there's a TODO somewhere about clarifying what they are.

Reply:
Gotcha. Seems like the tests still work after reverting my changes.